dbuf-lsp/src/navigation_handler/hover.rs

```diff
@@ -0,0 +1,165 @@
+/// Hover provider.
+///
+use tower_lsp::lsp_types::{LanguageString, MarkedString};
+
+use crate::core::ast_access::{ElaboratedHelper, File};
+use crate::core::dbuf_language::get_bultin_types;
+use crate::core::navigator::Symbol;
+use crate::core::pretty_printer::PrettyPrinter;
+
+pub fn get_hover(symbol: Symbol, file: &File) -> Vec<MarkedString> {
+    let elaborated = file.get_elaborated();
+
+    let mut strings = Vec::new();
```

Comment:
Instead of preparing `mut` vector in advance and pushing into it, you can just return their own result from each respective branch